modules/sfp_bevigil.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_bevigil
+# Purpose:      Search BeVigil OSINT API for subdomains.
+#
+# Author:      alt-glitch <https://github.com/alt-glitch>
+#
+# Created:     2022-08-27
+# Copyright:   (c) alt-glitch 2022
+# Licence:     MIT
+# -------------------------------------------------------------------------------
+
+import json
+import time
+
+from spiderfoot import SpiderFootEvent, SpiderFootPlugin
+
+
+class sfp_bevigil(SpiderFootPlugin):
+    meta = {
+        'name': "BeVigil API",
+        'summary': "Get Subdomains and URLs belonging to one target domain using OSINT Data made available by BeVigil",
+        'flags': ["apikey"],
+        'useCases': ["Passive", "Footprint"],
+        'categories': ["Crawling and Scanning"],
+        'dataSource': {
+            'website': "https://bevigil.com/osint-api",
+            'model': "FREE_AUTH_LIMITED",
+            'references': [
+                "https://bevigil.com/osint/api-documentation"
+            ],
+            'apiKeyInstructions': [
+                "Visit https://bevigil.com/osint-api",
+                "Click on 'Sign up for API Key'",
+                "Enter your email to get the OTP",
+                "Enter the OTP and Sign Up",
+                "Click on 'Get Started' to activate your account and you will have your API key"
+            ],
+            'favIcon': "https://bevigil.com/favicon.ico",
+            'description':"A powerful tool that provides access to millions of asset footprint "
+            "data points including domain intel, cloud services, API information, and third party "
+            "assets extracted from millions of mobile apps being continuously uploaded "
+            "and scanned by users on bevigil.com",
+        }
+    }
+
+    opts = {
+        'verify': True,
+        'api_key': '',
+        'delay': 1
+    }
+
+    optdescs = {
+        "verify": "DNS Resolve each identified subdomain.",
+        "api_key": "BeVigil API Key",
+        "delay": "Delay between requests, in seconds "
+    }
+
+    results = None
+    errorState = False
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+        self.__dataSource__ = "BeVigil OSINT API"
```

Comment:
This is not needed, since the module name will be used as the data source by default.

Reply:
Fixed